### PR TITLE
Traffic label transparent transmission plugin reconstruction

### PR DESCRIPTION
**Pull request by daizhenyu**
【Fix issue】#1360

[Modification content] The project structure of the traffic label transparent transmission plug-in has been modified, and the plug-in has been modularly reconstructed according to the component version granularity.

[Use case description] Not required

[Self-test situation] 1. Local static check passed

[Scope of influence] Not involved

### CODE REVIEW

File: sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.6.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/alibabadubbo/interceptors/AlibabaDubboConsumerInterceptor.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  *   limitations under the License.
  */
 
-package com.huaweicloud.sermant.tag.transmission.interceptors.rpc.dubbo;
+package com.huaweicloud.sermant.tag.transmission.alibabadubbo.interceptors;
```
Comment (luanwenfei-venus) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.6.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/alibabadubbo/interceptors/AlibabaDubboConsumerInterceptor.java:17]: 68行加一个注释解释为什么通过isEmpty判断，并且为什么put null

File: sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo3.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/dubbov3/interceptors/ApacheDubboConsumerInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * 向RpcInvocation中添加流量标签
+     *
+     * @param invocation Apache Dubbo 标签传递载体
+     */
+    @Override
+    protected void injectTrafficTag2Carrier(RpcInvocation invocation) {
+        for (String key : TrafficUtils.getTrafficTag().getTag().keySet()) {
```
Comment (TangLeDaily) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo3.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/dubbov3/interceptors/ApacheDubboConsumerInterceptor.java]: keySet改成entrySet

File: sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.6.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/alibabadubbo/interceptors/AlibabaDubboConsumerInterceptor.java (hunk 1 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.6.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/alibabadubbo/interceptors/AlibabaDubboConsumerInterceptor.java:17]: 已解决

**Review by TangLeDaily — DISMISSED** [review]

File: sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.7.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/apachedubbov2/interceptors/ApacheDubboConsumerInterceptor.java [hunk 3]
```diff
@@ -55,16 +61,22 @@ protected ExecuteContext doBefore(ExecuteContext context) {
...
-        for (String key : TrafficUtils.getTrafficTag().getTag().keySet()) {
+        for (Map.Entry<String, List<String>> entry : TrafficUtils.getTrafficTag().getTag().entrySet()) {
+            String key = entry.getKey();
             if (!TagKeyMatcher.isMatch(key)) {
                 continue;
             }
-            List<String> values = TrafficUtils.getTrafficTag().getTag().get(key);
+            List<String> values = entry.getValue();
+
+            // values不会为空字符串，用于判定values为null场景
```
Comment (luanwenfei-venus) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.7.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/apachedubbov2/interceptors/ApacheDubboConsumerInterceptor.java]: 这里解释的不够清楚，是因为在Provider端，将字符串都转换成了集合，空字符串转换成了null，所以这里通过集合来判断isEmpty，values在此场景必定为null

Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.7.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/apachedubbov2/interceptors/ApacheDubboConsumerInterceptor.java]: 并不是空字符串转为了null，consumer端传来的null值才会put null，其他情况都是put的list。

Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-dubbo2.7.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/apachedubbov2/interceptors/ApacheDubboConsumerInterceptor.java]: 已修改

**Review by luanwenfei-venus — APPROVED** [review]

**Review by TangLeDaily — APPROVED** [review]

**Pull request merged**